tests/unit/s2n_self_talk_session_id_test.c

```diff
@@ -355,6 +354,13 @@ int main(int argc, char **argv)
         /* Negotiate the handshake. */
         EXPECT_SUCCESS(s2n_negotiate(conn, &blocked));
 
+        /* Make sure the get_session_id and get_session_id_length APIs are
+         * working as expected */
+        EXPECT_EQUAL(s2n_connection_get_session_id_length(conn), 32);
+        session_id = malloc(s2n_connection_get_session_id_length(conn) + 1);
```

Comment:
Array on stack (or static memory)  might be a better choice for the test, as you don't have to worry about deallocating it